### PR TITLE
Refactor the file structure of the data package to be more consistent

### PR DESCRIPTION
**Pull request by nickrum**
## Scope

What's changed:

- There is now an index file for each folder, but all files in the package import from specific files directly to prevent circular dependencies between files
- Index files only contain re-exports

## Potential Risks / Drawbacks

- More files

## Review Questions

- Le me know, what you think!

### CODE REVIEW

**Review by jaads — APPROVED** [review]
nice 👍 makes sense to me! double checked with the dependency cruiser - looks all good :)

File: packages/data/src/types/abstract-query.ts [hunk 1]
```diff
@@ -34,9 +34,3 @@ export interface AbstractQuery {
  * @TODO
  * - Rethink every / some
  */
```
Comment (paescuj) [inline packages/data/src/types/abstract-query.ts:36]: @jaads Probably unrelated to this PR, but this comment feels also out of place

(https://github.com/directus/directus/pull/19941/files#diff-30f1d3b46baaef5d5605d82b0c67c65853cbd90f6092ee92563ca1b067359e35R10-R36)

File: packages/data/src/types/abstract-query/fields/related.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 /**
  * This file will be split up into multiple files soon.
  */
```
[the remark this answers is not in the document]
Comment (jaads) [reply packages/data/src/types/abstract-query/fields/related.ts:3]: Oh! No👍

**Review by paescuj — APPROVED** [review]
Looks clean to me 👍
@rijkvanzanten You happy too? 😃

**Comment by nickrum** [comment]
I guess this has already been merged as part of #19942. @rijkvanzanten merges faster than his shadow 😅

**Comment by paescuj** [comment]
> I guess this has already been merged as part of #19942.

The whole PR?

> @rijkvanzanten merges faster than his shadow 😅

Merge train, choo choo 😅

giphy

**Comment by nickrum** [comment]
> The whole PR?

Yeah, I split up the chore into its own PR to make reviewers life easier, based the type changes on the chore PR, but forgot to mark the other PR as draft.

**Pull request closed without merging**